src/lib/Transform.ts

```diff
@@ -220,9 +221,18 @@ export function batch<T>(
 				swallowErrors(earlyFlush());
 			}
 
-			throwIfThrowable(earlyFlushError);
-			throwIfThrowable(consumeEarlyFlushError()); // Catch the error early if possible
-			throwIfThrowable(flushFailureError);
+			const toThrow =
+				earlyFlushError ||
```

Comment:
Nitpick: use `??` instead of `||`. Otherwise good change, especially the detailed explanation.

Reply:
Did not know about `??`!

Reply:
Very cool

Reply:
Indeed :) I just noticed that I missed a few in previews reviews, btw, so perhaps check these too.
It's fairly recent stuff, so the rest of the codebase doesn't really use them either.

But especially because these 'errors' come from user code, and thus may actually not be errors, it's probably better to be as specific as possible.